src/components/layout/header.tsx

```diff
@@ -18,6 +20,17 @@ export const Header: React.FC = () => {
         </Link>
         <Flex sx={{ gap: 3, mt: 'auto', ml: 'auto', alignItems: 'flex-end' }}>
           {/** Navigation links go here */}
+          <Link
```

Comment:
Maybe use `NavLink`? Should also use `active` state with `.active` class instead, there's [an example on StackOverflow](https://stackoverflow.com/a/59410881/11499665).

This can be implemented in theme, like [how theme-ui does it for their own documentation](https://github.com/system-ui/theme-ui/blob/455cf08a1130e02c734906dab91aee256a19e049/packages/docs/src/gatsby-plugin-theme-ui/index.js#L160-L163).